src/main/java/aeneas/models/Board.java

```diff
@@ -11,20 +16,61 @@
   ArrayList<PlacedPiece> hints;
 
   public Board() {
+    pieces = new ArrayList<>();
+    hints = new ArrayList<>();
     for (int i=0;i<SIZE;i++){
       for (int j=0;j<SIZE;j++){
         squares[i][j] = true;
       }
     }
   }
+  
+  /**
+   * Count the number of valid squares on this board
+   * @return The number of valid squares.
+   */
+  int numValidSquares() {
+    int count = 0;
+    for(int j = 0; j < squares.length; j++) {
+      for(int i = 0; i < squares[j].length; i++) {
+         count += squares[j][i] ? 1 : 0;
+      }
+    }
+    return count;
+  }
+
+  /**
+   * Check if a square on the board is valid.
+   * If the row or col is < 0 or > 12, the square is invalid,
+   * otherwise check if the square is valid in the squares array.
+   * @return True if the square is valid, false otherwise.
+   */
+  boolean locationValid(Square s) {
```

Comment:
Not quite sure what you mean. If there is a square in the middle of the board that is not playable (i.e. squares[r][c] for that square is false), and you ask if the location is valid this will return false since squares[r][c] is false.